scripts/upd-kernel

```diff
@@ -56,20 +56,28 @@ popd > /dev/null
 # explode the initrd
 mkdir initrd
 pushd initrd > /dev/null
-zcat $OLDDIR/initrd.img | cpio -id --quiet
+xzcat $OLDDIR/initrd.img | cpio -id --quiet
 
-for mod in $(find modules/ -type f -name '*.ko.gz' -exec basename {} \; | sort | uniq) ; do
-    for path in $(find $WORKDIR/rpm/lib/modules/$NEWVER/ -name ${mod%.gz}); do
-        dest=${path##$WORKDIR/rpm/lib/}
+# delete the old module dir
+for oldmoddir in $(find lib/modules/ -type f -name 'modules.dep' -exec dirname {} \; ) ; do
+    rm -rf $oldmoddir
+done
+
+for mod in $(find lib/modules/ -type f -name '*.ko' -exec basename {} \; | sort | uniq) ; do
+    for path in $(find $WORKDIR/rpm/lib/modules/$NEWVER/ -name ${mod}); do
+        dest=${path##$WORKDIR/rpm/}
         mkdir -p $(dirname $dest)
-        gzip < $path > $dest.gz
+        cp -p $path $dest
     done
 done
 
+cp -p $WORKDIR/rpm/lib/modules/$NEWVER/modules.order lib/modules/$NEWVER
+cp -p $WORKDIR/rpm/lib/modules/$NEWVER/modules.builtin lib/modules/$NEWVER
+
 /sbin/depmod -a -b . $NEWVER
-rm -f modules/$NEWVER/modules.*map
+rm -f lib/modules/$NEWVER/modules.*map
 
-find . | cpio --quiet -c -o | gzip -9 > $NEWDIR/initrd.img
+find . | cpio --quiet -c -o | xz --format=lzma -9 > $NEWDIR/initrd.img
```

Comment:
Is the ``lzma`` format really needed here? (no idea) And I'd really like to see ``-T0`` (run in NUM_CPUS threads) used here.